File: design-patterns/go/mediator/general/colleague3.go
// colleague3.go

package main

import "fmt"

type Colleague3 struct {
	*Colleague
}

func NewColleague3(mediator *Mediator) (colleague3 *Colleague3) {
	colleague3 = &Colleague3{}
	colleague3.Colleague = NewColleague(mediator)
	return
}

func (colleague3 *Colleague3) ReceiveMessage(msg string) {
	fmt.Printf("Message received in Colleague3: %s\n",  msg)
}

func (colleague3 *Colleague3) SendMessage(colleague IColleague, msg string) {
	colleague3.mediator.SendMessage(colleague, msg)
}

File: design-patterns/ts/mediator/general/colleague3.ts
// colleague3.ts

import Colleague from "./colleague";
import Mediator from "./mediator";

class Colleague3 extends Colleague {
    constructor(mediator: Mediator) {
        super(mediator);
    }

    sendMessage(colleague: Colleague, msg: string): void {
        this.mediator.sendMessage(colleague, msg);
    }

    receiveMessage(msg: string): void {
        console.log("Message received in Colleague3: " + msg);
    }
}

export default Colleague3;